crates/ruff_python_parser/src/soft_keywords.rs

```diff
@@ -161,3 +207,19 @@ fn soft_to_name(tok: &Tok) -> Tok {
         name: name.to_owned(),
     }
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum Position {
+    /// The lexer is at the start of a logical line, i.e., the start of a simple or compound statement.
+    Statement,
+    /// The lexer is at the start of a simple statement, e.g., a statement following a semicolon
+    /// or colon, as in:
+    /// ```python
+    /// class Class: type X = int
+    /// ```
+    SimpleStatement,
+    /// The lexer is within brackets, with the given bracket nesting depth.
+    Nested(u32),
+    /// The lexer is some other location.
+    Other,
```

Comment:
This and `Nested` could be combined, since this is just `Nested` with a depth of `0`. Alternatively, `Nested` could be `NonZeroU32`. But both felt clunkier in practice. Open to changing, though.